Review Committers and PMC Members docs (#60744)
mistercrunch/Airflow: COMMITTERS.rst

Author: This note is added here to prevent accidental merge. Please wait for merging until PMC vote is completed.

PMC voting end on [2026-01-26](https://www.timeanddate.com/counters/customcounter.html?year=2025&month=12&day=22)

accept other contributions. There is no strict protocol for becoming a committer. Candidates for new
committers are typically people that are active contributors and community members.

Some people might be active in several of those areas and while they might not have enough 'achievements' in any
single one of those, their combined contributions in several areas all count.

As a community, we appreciate contributions to the Airflow codebase, but we also place equal value
on those who help Airflow by improving the community in some way. It is entirely possible to become

> a committer (and eventually a PMC member) without ever having to change a single line of code,
> though that requires visibility and presence in the regular community channels - slack, devlist, social
> media - this is important for the PMC to be aware of such activities. If PMC is not aware of those, they

Reviewer: ```suggestion
a committer (and eventually a PMC member) without ever having to change a single line of code,
but that requires visibility and presence in the regular community channels - slack, devlist, social
media - this is important for the PMC to be aware of such activities. If PMC is not aware of those, they
```

> have no chance to spot and discuss such candidates.

Reviewer: ```suggestion
have no chance to spot and promote such candidates.
```

Guidelines from ASF are listed at ASF:
`New Candidates for Committership <http://community.apache.org/newcommitter.html#guidelines-for-assessing-new-candidates-for-committership>`_

Prerequisites
^^^^^^^^^^^^^

General prerequisites that we look for in all candidates:

1.  Consistent contribution over at least three months
2.  Visibility on discussions on the dev mailing list, Slack channels or GitHub issues/discussions

>     including casting non-binding votes and testing RCs

Reviewer 2: ```suggestion
    including casting non-binding votes and testing release candidates.
```

Author: Inidrectly applied by other comment. Thanks for this!

>     including casting non-binding votes and testing RCs

Reviewer: ```suggestion
    including casting non-binding votes and testing release candidates for various Airflow releases.
```

> 3.  Helping other contributors and users - with sharing their code contributor's experience and reviewing other's code

Reviewer: This one isn't too clear to me. What are we intending to convey in: `with sharing their code contributor's experience`?

Author: It means that it is not solely intended to drop code to be a committer but that you also engage reviewing others contributions. And providing constructive feedback that others can learn.
It means you bring-in your expertise to help others to benefit the project.

Do you have a proposal for wording in this?

Reviewer: ```suggestion
3.  Helping other contributors and users by engaging in their code contributor's experience like providing constructive feedback, suggestions, and reviewing their code
```

Why not it literally :)

> 4.  Contributions to community health and project's sustainability for the long-term

Reviewer: ```suggestion
4.  Contributions to community health and project's sustainability in the long-term
```

5.  Understands contributor/committer guidelines: `Contributors' Guide <contributing-docs/README.rst>`__

Code contribution
^^^^^^^^^^^^^^^^^

> A prospective committer typically demonstrates several of the following:

Reviewer: ```suggestion
A potential candidate for committership typically demonstrates several of the following:
```

Reviewer: IDK, sounds clearer?

1.  Makes high-quality commits (including meaningful commit messages), and assess the impact of the changes, including
    upgrade paths or deprecation policies
2.  Tests Release Candidates to help the release cycle

> 3.  Proposes and leads Airflow Improvement Proposal(s) to completion

Reviewer: ```suggestion
3.  Proposes and/or leads Airflow Improvement Proposal(s) to completion
```

    - Docker Image
    - Helm Chart
    - Dev Tools (Breeze / CI)

>     - Providers - especially those where PMC is a steward

Reviewer 3: AIP-95 is yet to be approved, so I'd either remove it for now, or add a future reference to what a steward is (so it will be applied immediately as the AIP is approved)

Author: Has been voted on, see https://lists.apache.org/thread/24ryxox0z1z9opkfp57b6z91j0khlqso

...but I assume changes in process are not implemented (yet).

Reviewer 4: It's already approved :) 

https://lists.apache.org/thread/24ryxox0z1z9opkfp57b6z91j0khlqso

5.  Has made a significant improvement or added an integration with services/technologies important to the Airflow
    Ecosystem

> 6.  Actively participated in the security process, as a member of the security team, discussing, assessing and

Reviewer: This alone is in the past. Maybe have it in same _tense_ as others

Author: You mean too much looking into past. How ybout... or do you have a different proposal for phrasing?
```suggestion
6.  Actively participates in the security process, as a member of the security team, discussing, assessing and
```

Reviewer: What you have is a good enough proposal

    fixing security issues.

> 7.  Demonstrates skills to maintain the complex codebase to ensure keeping maturity of product and complexity

Reviewer: Like CI?

Author: Yes. As well as other complexity we have, e.g. refactoring pytests to make them modular, better readable or more stable.

8.  Has been actively helping other contributors and users - in Slack, GitHub issues, GitHub discussions - by
    guiding their coding practices, reviewing their code and answering code-related questions

Note: As mentioned earlier, there may be exceptions for code contributions requirements for contributors who

> provide sustained, high-impact non-code contributors (for example, community leadership, outlining AIPs, etc.).

Reviewer 3: ```suggestion
provide sustained, high-impact non-code contributions (for example, community leadership, outlining AIPs, etc.).
```

> provide sustained, high-impact non-code contributors (for example, community leadership, outlining AIPs, etc.).

Reviewer 3: ```suggestion
provide sustained, high-impact non-code contributions (for example, community leadership, outlining AIPs, etc.).
```

Author: Thansk for the typo correction!

Such exceptions are rare and chosen selectively, so contributors with coding skills who aim to become committers

> and PMC are generally encouraged to follow the regular path.

Reviewer 2: ```suggestion
and PMC members are generally encouraged to follow the regular path.
```

  the case if no one from the community is taking it on.
* Improve processes and tooling
* Refactoring code
* Taking part in handling regular chores - i.e. maintenance of Airflow - for example making sure that
  our canary builds are green, upgrading dependencies regularly, monitoring for security alerts.
  Ideally also automating those as much as possible.

Guidelines for promoting Committers to Airflow PMC
--------------------------------------------------

To become a PMC member the committers should meet all **general prerequisites**.
Apart from that the person should demonstrate distinct and sustained **community involvement** or
**code contributions**.

Prerequisites
^^^^^^^^^^^^^

* Has been a committer for at least 3 months
* Is currently an active community member

>   (Visible in mailing list, other channel discussions or reviewing PRs at the minimum)

Reviewer: This seems repetitive, we mention it below in "community involvement"

Author: Yes this is the intend. (In my view)

Reviewer: Cool

Community involvement
^^^^^^^^^^^^^^^^^^^^^

* Visibility on discussions on the dev mailing list
* Spreading the word for "Airflow" either:

  * Talks at meetups, conferences, etc
  * Creating content like videos, blogs, etc

>   * Actively participates in public discussions about Airflow happening in social media or Github,

Reviewer 2: ```suggestion
  * Actively participates in public discussions about Airflow happening in social media or GitHub,
```

    promoting Airflow as a product

* Growing the community:

  * Mentors new members/contributors
  * Answers users/contributors via GitHub issues, dev list or slack

> * Is aware and pays attention to Airflow public image

Reviewer: ```suggestion
* Is aware and pays attention to public image of Airflow
```

  * Watches and takes part in public discussions about Airflow
  * Flags cases which might be problematic for Airflow community
  * Is wary about the use of Airflow Registered trademark and name

Code contribution
^^^^^^^^^^^^^^^^^

> * Consistent voting on RCs for at least past 3 releases lifecycles

Author: Applying suggestion from Wei here as well.
```suggestion
* Consistent voting on release candidates for at least past 3 releases lifecycles
```

The PMC also can mark committers as inactive after they have not been involved in the community for
more than 12 months.

New Committer Onboarding Steps
------------------------------

To be able to merge PRs, committers have to integrate their GitHub ID with Apache systems. To do that follow steps:

1.  Verify you have a GitHub ID
    `enabled with 2FA <https://help.github.com/articles/securing-your-account-with-two-factor-authentication-2fa/>`__.
2.  Merge your Apache and GitHub accounts using
    `GitBox (Apache Account Linking utility) <https://gitbox.apache.org/setup/>`__. This also asks you to link your
    GitHub ID to your Apache account. You should see 5 green checks in GitBox.
3.  Wait at least 30  minutes for an email inviting you to Apache GitHub Organization and accept invitation.
4.  After accepting the GitHub Invitation verify that you are a member of the
    `Airflow committers team on GitHub <https://github.com/orgs/apache/teams/airflow-committers>`__.

   Additionally, as a committer you can view the team membership at:

   * https://github.com/orgs/apache/teams/airflow-committers/members
   * https://whimsy.apache.org/roster/committee/airflow

> 5.  Ask in ``#internal-airflow-ci-cd`` channel to be

Reviewer 5: @jscheffl , I recall this is not needed anymore? @bugraoz93 correct me if I'm wrong?

Author: Yes, at the moment self-hosted runners are not in use... but who knows. Maybe in future we need to re-activate if we have a stability, capacity or performance issue... or self-hosted runners are much cooler.

As long as GH does have native runners in sufficient speed and amount, we keep this as spare to say. (Any save money/funding)

    by the CI team. Wait for confirmation that this is done and some helpful tips from the CI team (Temporarily disabled)
6.  After confirming that step 5 is done, open a PR to include your GitHub ID in:

    * ``dev/breeze/src/airflow_breeze/global_constants.py`` (COMMITTERS variable)
    * name and GitHub ID in `project.rst <https://github.com/apache/airflow/blob/main/airflow-core/docs/project.rst>`__.
    * If you had been a collaborator role before getting committer, remove your GitHub ID from ``.asf.yaml``.

7.  Raise a PR to `airflow-site <https://github.com/apache/airflow-site>`_ repository with the following additions:
    (A kind request: If there are other committers who joined around the same time, please create a unified PR for
    all of you together.)

    * List your name(s) in the
      `committers list <https://github.com/apache/airflow-site/blob/main/landing-pages/site/data/committers.json>`__.
    * Post an entry in
      `Announcements <https://github.com/apache/airflow-site/blob/main/landing-pages/site/content/en/announcements/_index.md>`__.

8.  Ask a release manager to make Slack announcement

> New PMC Onboarding steps

Reviewer 2: ```suggestion
New PMC Member Onboarding steps
```

> ------------------------

Reviewer 6: ```suggestion
New PMC Member Onboarding steps
-------------------------------------
```

Author: Almost...
```suggestion
New PMC Member Onboarding steps
-------------------------------
```

> 1.  Familiarise yourself with https://community.apache.org/pmc/responsibilities.html

Reviewer 2: ```suggestion
1.  Familiarise yourself with `<PMC Responsibilities>https://community.apache.org/pmc/responsibilities.html`_
```

> 2.  Subscribe to the private mailing list: private@airflow.apache.org. Do this by sending an empty email to

Reviewer 2: ```suggestion
2.  Subscribe to the private mailing list: ``private@airflow.apache.org``. Do this by sending an empty email to
```

>     private-subscribe@airflow.apache.org and following the instructions in the automated response you'll receive.

Reviewer 2: ```suggestion
    ``private-subscribe@airflow.apache.org`` and following the instructions in the automated response you'll receive.
```
